vfs/memdb/api.go

```diff
@@ -66,3 +69,18 @@ func Delete(name string) {
 	defer memoryMtx.Unlock()
 	delete(memoryDBs, name)
 }
+
+// TestDB creates an empty shared memory database for the test to use.
+// The database is automatically deleted when the test and all its subtests complete.
+// Each subsequent call to TestDB returns a unique database.
+func TestDB(tb testing.TB) string {
+	tb.Helper()
+	name := fmt.Sprintf("%s_%p", tb.Name(), tb)
+	tb.Cleanup(func() { Delete(name) })
+	Create(name, nil)
```

Comment:
I did so, because I think it has better (more robust) semantics.

There are two kinds of `memdb`s: databases that start with `/` are shared, others are private.

Private databases are easy, each connection gets a new instance, regardless of the name.

Shared databases are reference counted: the first time you open a connection with a certain "path" (e.g. `/abc`) you get a new database; then, as long as there is always at least one connection open, you always get the same database for that "path" (e.g. `/abc`). But as soon as you close the last connection to that database, the reference count reaches zero, and it gets deleted.

None of this is really subject to change, as these semantics are for compatibility with [SQLite's `memdb`](https://sqlite.org/src/doc/tip/src/memdb.c).

So, if for any reason you close all connections to the database in the middle of the test, the database is deleted. Which I'd find surprising (and unlike `t.TempDir()`), particularly with `database/sql` as you don't manage the connection pool.

Doing an explict `Create` and `Delete` avoids this.

Reply:
Ohh. I see. So that then also means you could have multiple tests run in parallel which each `Open()` and `Close()` the same DB and see the same data, as long as that one is created outside of the subtests.